e2e_playwright/st_dataframe_selections_test.py

```diff
@@ -0,0 +1,310 @@
+    canvas = _get_multi_row_and_column_select_df(app)
+
+    # toolbar has three buttons: download, search, fullscreen
+    data_editor_toolbar = canvas.get_by_test_id("stElementToolbar")
+    toolbar_buttons = data_editor_toolbar.get_by_test_id("stElementToolbarButton")
+    expect(toolbar_buttons).to_have_count(3)
+
+    _select_some_rows_and_columns(app, canvas)
+    _expect_multi_row_multi_column_selection(app)
+    # toolbar has one more button now: clear selection
+    toolbar_buttons = data_editor_toolbar.get_by_test_id("stElementToolbarButton")
+    expect(toolbar_buttons).to_have_count(4)
+    # click on the clear-selection button which is the first in the toolbar
+    toolbar_buttons.nth(0).click()
+    wait_for_app_run(app)
+    expected = "Dataframe multi-row-multi-column selection: {'select': {'rows': [], 'columns': []}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+
+# def test_in_form_selection_and_session_state(app: Page):
+#     canvas = _get_in_form_df(app)
+#     _select_some_rows_and_columns(app, canvas)
+
+#     # nothing should be shown yet because we did not submit the form
+#     expected = "Dataframe-in-form selection in session state: {'select': {'rows': [], 'columns': []}}"
+#     selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+#     expect(selection_text).to_have_count(1)
+
+#     app.wait_for_timeout(500)
+#     # submit the form
+#     app.get_by_test_id("stFormSubmitButton").first.locator("button").click()
+#     wait_for_app_run(app)
+#     # expected =  "Dataframe-in-form selection in session state: {'select': {'rows': [0, 2], 'columns': ['col_1', 'col_3', 'col_4']}}"
+#     expected = "Dataframe-in-form selection in session state:"
+#     selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+#     # wait_until(
+#     #     app,
+#     #     lambda: app.get_by_test_id("stMarkdownContainer")
+#     #     .filter(has_text=expected)
+#     #     .count()
+#     #     == 1,
+#     # )
+
+#     expect(selection_text).to_have_count(1)
+#     expect(app.get_by_test_id("stMarkdownContainer").nth(13)).to_have_text(
+#         "Dataframe-in-form selection in session state: {'select': {'rows': [0, 2], 'columns': ['col_1', 'col_3', 'col_4']}}",
+#         timeout=2000,
+#     )
+
+
+def test_in_form_selection_and_session_state(app: Page):
+    canvas = _get_in_form_df(app)
+    _select_some_rows_and_columns(app, canvas)
+
+    # nothing should be shown yet because we did not submit the form
+    expected = "Dataframe-in-form selection: {'select': {'rows': [], 'columns': []}}"
+    selection_text = app.get_by_test_id("stMarkdownContainer").filter(has_text=expected)
+    expect(selection_text).to_have_count(1)
+
+    # submit the form. Without the timeout the test is flaky, presumably because the submit-button does not have an eventlistener on time :/
```

Comment:
@raethlein It might be because of the 150ms debounce, maybe its better to use 200ms to make sure that its guaranteed to be longer then the debounce.